# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: testing/web-platform/tests/editing/crashtests/delete-from-ruby-at-start-of-button.html
# Commit: 6d9a0abd0a3c
# Full Hash: 6d9a0abd0a3c26f4337aff723c15795fe91fe884
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2023-12-27 09:19:35
# Description:
#   Bug 1870962 - Make `HTMLEditor::SplitAncestorStyledInlineElementsAt` never drop the "did split" state from the result r=m_kato
#   
#   Oddly, it returns the latest result as split result. However, in the case,
#   it splits `<kbd>` under `<button>`, but does not split `<button>`, but the
#   caller expects that whether it did split or not.
# ==============================================================================

diff -r 45dc475cd374 -r 6d9a0abd0a3c testing/web-platform/tests/editing/crashtests/delete-from-ruby-at-start-of-button.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/testing/web-platform/tests/editing/crashtests/delete-from-ruby-at-start-of-button.html	Wed Dec 27 02:53:44 2023 +0000
@@ -0,0 +1,29 @@
+<!doctype html>
+<html>
+<head>
+<meta charset="utf-8">
+<style>
+*:last-child {
+  white-space: pre-wrap;
+}
+</style>
+<script>
+addEventListener("DOMContentLoaded", () => {
+  getSelection().collapse(document.querySelector("ruby"));
+  document.designMode = "on";
+  document.execCommand("delete");
+});
+</script>
+</head>
+<body>
+<label contenteditable="true"></label>
+<bdi style="white-space: nowrap">
+<button>
+<kbd>
+<ruby>
+</ruby>
+</kbd>
+</button>
+</bdi>
+</body>
+</html>
